test/dataset.py

```diff
@@ -0,0 +1,153 @@
+import uuid
+
+NUM_CHILDREN_PER_PING = 3
+SCALAR_VALUE = 42
+
+ping_dimensions = {
+    "submission_date":   [u"20160305", u"20160607"],
+    "activity_date":     [u"2016-03-02T08:10:03.503Z", u"2016-06-01T08:10:17.492Z"],
+    "application":       [u"Firefox", u"Fennec"],
+    "doc_type":          [u"main", u"crash"],
+    "channel":           [u"nightly", u"aurora"],
+    "build_version":     [u"45.0a1", u"45"],
+    "build_id":          [u"20160301000000", u"20160302000000"],
+    "os_name":           [u"Linux", u"Windows_NT"],
+    "os_version":        [u"6.1", u"3.1.12"],
+    "architecture":      [u"x86", u"x86-64"],
+    "e10s":              [True, False],
+    "country":           ["US", "UK"],
+    "experiment_id":     [None, "displayport-tuning-nightly@experiments.mozilla.org"],
+    "experiment_branch": ["control", "experiment"],
+}
+
+def generate_pings():
+    for submission_date in ping_dimensions["submission_date"]:
+        for activity_date in ping_dimensions["activity_date"]:
+            for application in ping_dimensions["application"]:
+                for doc_type in ping_dimensions["doc_type"]:
+                    for channel in ping_dimensions["channel"]:
+                        for build_version in ping_dimensions["build_version"]:
+                            for build_id in ping_dimensions["build_id"]:
+                                for os_name in ping_dimensions["os_name"]:
+                                    for os_version in ping_dimensions["os_version"]:
+                                        for architecture in ping_dimensions["architecture"]:
+                                            for e10s in ping_dimensions["e10s"]:
+                                                for country in ping_dimensions["country"]:
+                                                    for experiment_id in ping_dimensions["experiment_id"]:
+                                                        for experiment_branch in ping_dimensions["experiment_branch"]:
+                                                            dimensions = {
+                                                                u"submission_date": submission_date,
```

Comment:
This for loop looks pretty crazy; maybe you could code a generic dimensions generator that takes as input `ping_dimensions`.